src/test/java/core/basesyntax/services/DataProcessorServiceTest.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.services.impl.DataProcessorServiceImpl;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class DataProcessorServiceTest {
+    private static DataProcessorService dataProcessorService;
+
+    @BeforeAll
+    static void beforeAll() {
+        dataProcessorService = new DataProcessorServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.STORAGE.clear();
+    }
+
+    @Test
+    void processData_validData_ok() {
+        List<String> data = List.of("b,banana,10", "b,apple,20", "s,banana,10");
+        List<FruitTransaction> expected = List.of(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 10),
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "apple", 20),
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 10));
+        List<FruitTransaction> fruitTransactions = dataProcessorService.processInputData(data);
+        assertEquals(expected, fruitTransactions);
+    }
+
+    @Test
+    void processData_invalidQuantity_notOk() {
+        List<String> data = List.of("b,banana,-200", "b,apple,20", "s,banana,10");
+        assertThrows(RuntimeException.class, () -> dataProcessorService.processInputData(data));
+    }
+
+    @Test
+    void processData_invalidOperationType_notOk() {
+        List<String> data = List.of("buy, banana,-200", "b,apple,20", "s,banana,10");
+        assertThrows(RuntimeException.class, () -> dataProcessorService.processInputData(data));
+    }
```

Comment:
there is no reason to handle some general error, if you want this case to be handled by your code, please create you own exception throw it with some error message and check in tests

Reply:
Not fixed

Reply:
I try this and get such error 
DataProcessorServiceTest.processData_invalidOperationType_notOk:48 Unexpected exception type thrown, expected: <core.basesyntax.exception.InvalidOperationTypeException> but was: <java.lang.RuntimeException>